src/screens/TransactionDetail.js

```diff
@@ -49,45 +55,71 @@ class TransactionDetail extends React.Component {
   /**
    * Get accumulated weight and confirmation level of the transaction
    */
-  getConfirmationData = () => {
-    hathorLib.txApi.getConfirmationData(this.props.match.params.id, (data) => {
-      this.setState({ confirmationData: data });
-    }, (e) => {
+  getConfirmationData = async () => {
+    this.setState({
+      confirmationData: null,
+      confirmationDataError: false,
+    });
+
+    try {
+      const data = await this.props.wallet.getTxConfirmationData(this.props.match.params.id);
+      this.setState({
+        confirmationData: data,
+        confirmationDataError: false,
+      });
+    } catch(e) {
       // Error in request
       console.log(e);
-    });
-  }
 
-  /**
-   * Update state after receiving the transaction response back from the server
-   */
-  txReceived(data) {
-    if (data.success) {
-      this.setState({ transaction: data.tx, meta: data.meta, spentOutputs: data.spent_outputs, loaded: true, success: true });
-    } else {
-      this.setState({ loaded: true, success: false, transaction: null });
+      this.setState({
+        confirmationData: null,
+        confirmationDataError: true,
+      });
     }
   }
 
   /**
    * Get transaction in the server when mounting the page
    */
-  getTx() {
-    hathorLib.txApi.getTransaction(this.props.match.params.id, (data) => {
-      this.txReceived(data);
-      if (data.success && !hathorLib.helpers.isBlock(data.tx)) {
+  async getTx() {
+    this.setState({
+      loaded: false,
+      transaction: null,
+      meta: null,
+      spentOutputs: null,
+      success: null,
+    });
+
+    try {
+      const data = await this.props.wallet.getFullTxById(this.props.match.params.id);
+
+      if (!hathorLib.helpers.isBlock(data.tx)) {
         this.getConfirmationData();
       }
-    }, (e) => {
+
+      this.setState({
```

Comment:
If you use an id that does not exist in the full node, `data.success` will be `false`. In that case, we should show "Transaction not found" on the screen. Otherwise, in case of an API error, we can show "Error retrieving transaction, try again".

What do you think?

Reply:
The problem here is that the lib handles the `success: false` by throwing an error, that is not `Transaction not found`

To do this, we need to refactor the lib to return a standard message on both facades so we can handle in the clients

Do you think we should do this now?

Reply:
@pedroferreira1 Done in f56a6131d919ccc9fed577841a0798bdfe4effdf, which depends on the [new lib PR](https://github.com/HathorNetwork/hathor-wallet-lib/pull/456)